app/src/main/java/com/nextcloud/talk/jobs/SaveFileToStorageWorker.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Nextcloud Talk application
+ *
+ * @author Andy Scherzinger
+ * @author Marcel Hibbe
+ * Copyright (C) 2022 Andy Scherzinger <info@andy-scherzinger.de>
+ * Copyright (C) 2021 Marcel Hibbe <dev@mhibbe.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.nextcloud.talk.jobs
+
+import android.content.Context
+import android.util.Log
+import androidx.work.Data
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import autodagger.AutoInjector
+import com.nextcloud.talk.application.NextcloudTalkApplication
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+
+
+@AutoInjector(NextcloudTalkApplication::class)
+class SaveFileToStorageWorker(val context: Context, workerParameters: WorkerParameters) :
+    Worker(context, workerParameters) {
+
+    private var totalFileSize: Long = -1
+
+    override fun doWork(): Result {
+        if (totalFileSize > -1) {
+            setProgressAsync(Data.Builder().putInt(PROGRESS, 0).build())
+        }
+
+        try {
+            val fileName = inputData.getString(KEY_FILE_NAME)
+            val sourceFilePath = inputData.getString(KEY_SOURCE_FILE_PATH)
+
+            val cacheFile = File(sourceFilePath)
+
+            val downloadDirectory = "storage/self/primary/Download"
```

Comment:
i think it would be better to use MediaStore to store the files.
https://developer.android.com/training/data-storage/shared